pkg/containerd/opts/container.go

```diff
@@ -71,13 +71,21 @@ func WithVolumes(volumeMounts map[string]string) containerd.NewContainerOpts {
 		if err != nil {
 			return err
 		}
-		defer os.RemoveAll(root) // nolint: errcheck
-		for _, m := range mounts {
-			if err := m.Mount(root); err != nil {
-				return err
-			}
+		// We change RemoveAll to Remove so that we either leak a temp dir
+		// if it fails but not RM snapshot data.
+		// refer to https://github.com/containerd/containerd/pull/1868
+		// https://github.com/containerd/containerd/pull/1785
+		defer os.Remove(root) // nolint: errcheck
+		if err := mount.All(mounts, root); err != nil {
+			return errors.Wrap(err, "failed to mount")
 		}
-		defer unix.Unmount(root, 0) // nolint: errcheck
+		defer func() {
```

Comment:
I prefer we log the error here in our code.
Or else if `err != nil`, we'll lose the error info.

Reply:
Yes. It is same with happened in containerd. When i modify it in containerd, I don't know how to aggregate these errors. So...
I will find the way to aggregate it again.

Reply:
I think log it is fine. :)

Reply:
done